subprojects/client-protocol/src/main/java/org/gradle/profiler/client/protocol/agent/AgentClientContainer.java

```diff
@@ -0,0 +1,23 @@
+package org.gradle.profiler.client.protocol.agent;
+
+import org.gradle.profiler.client.protocol.Client;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+public enum AgentClientContainer {
```

Comment:
Can we maybe get rid of this client "holder" altogether and pass around the client after it has been created?

Reply:
This container is used so the client is shared between agent and interceptor. I think it is possible, but it would require some extra effort to modify instrumentation. I think it's easier with a static reference. WDYT?

Reply:
Ok, then let's leave it as is.